bin/packages/watch.js

```diff
@@ -26,63 +25,55 @@ const exists = ( filename ) => {
 // Exclude test files including .js files inside of __tests__ or test folders
 // and files with a suffix of .test or .spec (e.g. blocks.test.js),
 // and deceitful source-like files, such as editor swap files.
-const isSourceFile = ( filename ) => {
-	return (
+const isSourceFile = ( filename, skip ) => {
+	const isSource =
+		/.\.(js|json|scss)$/.test( filename ) &&
+		/\/src\/.+/.test( filename ) &&
 		! [
-			/\/(benchmark|__mocks__|__tests__|test|storybook|stories)\/.+.js$/,
+			/\/(benchmark|__mocks__|__tests__|test|storybook|stories)\/.+/,
 			/.\.(spec|test)\.js$/,
-		].some( ( regex ) => regex.test( filename ) ) &&
-		/.\.(js|json|scss)$/.test( filename )
-	);
+		].some( ( regex ) => regex.test( filename ) );
+
+	return isSource ? true : skip;
```

Comment:
This should be a nice improvement, as `skip` avoids watching non-matching subfolders entirely.